tools/SetupFlow/DevHome.SetupFlow/Services/WinGet/WinGetPackageInstaller.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using System;
+using System.Threading.Tasks;
+using DevHome.SetupFlow.Common.Extensions;
+using DevHome.SetupFlow.Common.Helpers;
+using DevHome.SetupFlow.Common.WindowsPackageManager;
+using DevHome.SetupFlow.Exceptions;
+using DevHome.SetupFlow.Models;
+using Microsoft.Management.Deployment;
+using Windows.Win32.Foundation;
+
+namespace DevHome.SetupFlow.Services.WinGet;
+
+/// <summary>
+/// Installs a package using the Windows Package Manager (WinGet).
+/// </summary>
+internal class WinGetPackageInstaller : IWinGetPackageInstaller
+{
+    private readonly WindowsPackageManagerFactory _wingetFactory;
+    private readonly IWinGetPackageFinder _packageFinder;
+
+    public WinGetPackageInstaller(WindowsPackageManagerFactory wingetFactory, IWinGetPackageFinder packageFinder)
+    {
+        _wingetFactory = wingetFactory;
+        _packageFinder = packageFinder;
+    }
+
+    /// <inheritdoc />
+    public async Task<InstallPackageResult> InstallPackageAsync(WinGetCatalog catalog, string packageId)
+    {
+        if (catalog == null)
+        {
+            throw new CatalogNotInitializedException();
+        }
+
+        // 1. Find package

```

Comment:
Optimize this by caching the OOP object safely (without proxy object leakage which can cause unexpected RPC errors)